workflow/controller/operator.go

```diff
@@ -1365,14 +1365,62 @@ func getTemplateOutputsFromScope(tmpl *wfv1.Template, scope *wfScope) (*wfv1.Out
 	return &outputs, nil
 }
 
+func checkVariableRefInTmpl(template *wfv1.Template, variable string) bool {
```

Comment:
Seems like this function is only called in `hasOutputResultRef`. Don't see the need for a separate function for this. Can we combine it?